### PR TITLE
SONARJAVA-3726 Update documentation to support SQ 8.8

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: docs/CUSTOM_RULES_101.md [hunk 1]
```diff
@@ -41,39 +41,29 @@ A custom plugin is a Maven project, and before diving into code, it is important
...
 * `pom.xml`: use a snapshot version of the Java Analyzer
-* `pom_SQ_7_9_LTS.xml`: self-contained `pom` file, configured with dependencies matching SonarQube 7.9 LTS requirements
-* `pom_SQ_8_7.xml`: self-contained `pom` file, configured with dependencies matching SonarQube 8.7
+* `pom_SQ_7_9_LTS.xml`: self-contained `pom` file, configured with dependencies matching SonarQube `7.9 LTS` requirements
+* `pom_SQ_8_8.xml`: self-contained `pom` file, configured with dependencies matching SonarQube `8.8`
 
-These 3 `pom`s correspond different use-cases, depending of which instance of SonarQube you will target with your custom-rules plugin. In this tutorial, **we will only use the file named `pom_SQ_8_7.xml`**, as it is completely independent from the build of the Java Analyzer, is self contained, and will target the latest release of SonarQube.
+These 3 `pom`s correspond different use-cases, depending of which instance of SonarQube you will target with your custom-rules plugin. In this tutorial, **we will only use the file named `pom_SQ_8_8.xml`**, as it is completely independent from the build of the Java Analyzer, is self contained, and will target the latest release of SonarQube.
```
Comment (dorian-burihabwa-sonarsource) [inline docs/CUSTOM_RULES_101.md:47]: `depending of` -> `depending on`
`self contained` -> `self-contained`

File: docs/CUSTOM_RULES_101.md [hunk 2]
```diff
@@ -41,39 +41,29 @@ A custom plugin is a Maven project, and before diving into code, it is important
...
 
 Let's start by building the custom-plugin template by using the following command:
 
 ```
-mvn clean install -f pom_SQ_8.7.xml
+mvn clean install -f pom_SQ_8.8.xml
 ```
 
-Note that you can also decide to **delete** the original pom.xml file (**NOT RECOMMANDED**), and then rename `pom_SQ_8.7.xml` into `pom.xml`. You would then be able to use the very simple command:
+Note that you can also decide to **delete** the original pom.xml file (**NOT RECOMMANDED**), and then rename `pom_SQ_8.8.xml` into `pom.xml`. You would then be able to use the very simple command:
```
Comment (dorian-burihabwa-sonarsource) [inline docs/CUSTOM_RULES_101.md:55]: `RECOMMANDED` -> `RECOMMENDED`

File: docs/CUSTOM_RULES_101.md [hunk 3]
```diff
@@ -515,10 +505,10 @@ Then, grab the jar file `java-custom-rules-example-1.0.0-SNAPSHOT.jar` from the
 >
 > :exclamation: **SonarQube Java Plugin compatible version**
 >
-> Before going further, be sure to have the adequate version of the SonarQube Java Analyzer with your SonarQube instance. The dependency over the Java Analyzer of our custom plugin is defined in its `pom`, as seen in the first chapter of this tutorial. We consequently provide two distinct `pom` files mapping both the `7.9` LTS version of SonarQube, as well as its latest release, version `8.7`.
+> Before going further, be sure to have the adequate version of the SonarQube Java Analyzer with your SonarQube instance. The dependency over the Java Analyzer of our custom plugin is defined in its `pom`, as seen in the first chapter of this tutorial. We consequently provide two distinct `pom` files mapping both the `7.9` LTS version of SonarQube, as well as its latest release, version `8.8`.
 >
 > * If your instance is SonarQube `7.9` LTS version, make sure to update the Java Analyzer to its latest compatible version through the SonarQube marketplace (it should be version version `6.3.2.22818`), and then use this the file `pom_SQ_7_9_LTS.xml` file to build the project.
```
Comment (dorian-burihabwa-sonarsource) [inline docs/CUSTOM_RULES_101.md:510]: `version version` -> `version`

File: docs/CUSTOM_RULES_101.md [hunk 4]
```diff
@@ -486,14 +476,14 @@ class MyJavaFileCheckRegistrarTest {
...
-Start by building the project using maven. Note that here we are using the self-contained `pom` file targeting SonarQube `8.7`. If you renamed it into `pom.xml`, remove the `-f pom_SQ_8_7.xml` part of the following command):
+Start by building the project using maven. Note that here we are using the self-contained `pom` file targeting SonarQube `8.8`. If you renamed it into `pom.xml`, remove the `-f pom_SQ_8_8.xml` part of the following command):
 
 
 ```
 $ pwd
 /home/gandalf/workspace/sonar-java/docs/java-custom-rules-example
   
-$ mvn clean install -f pom_SQ_8.7.xml
+$ mvn clean install -f pom_SQ_8.8.xml
```
Comment (dorian-burihabwa-sonarsource) [inline docs/CUSTOM_RULES_101.md:486]: Isn't there a step missing here about generating the custom rule's metadata files?

**Review by dorian-burihabwa-sonarsource — COMMENTED** [review]
LGTM. Just a few minor changes and a question about the `mvn clean install` command failing if the metadata files are not generated.
https://github.com/SonarSource/sonar-java/blob/SONARJAVA-3726/docs/java-custom-rules-example/README.md should also be updated with reference to the current version

File: docs/CUSTOM_RULES_101.md (hunk 4 above)
Comment (m-g-sonar) [reply docs/CUSTOM_RULES_101.md:486]: nop, if you use all the parameters of the annotation, you don't need the metadata files I think

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]

**Pull request merged**